api/odl.py

```diff
@@ -796,28 +803,44 @@ def _detail_for_elementtree_entry(cls, parser, entry_tag, feed_url=None, do_get=
         odl_license_tags = parser._xpath(entry_tag, 'odl:license') or []
         for odl_license_tag in odl_license_tags:
             identifier = subtag(odl_license_tag, 'dcterms:identifier')
-            content_type = subtag(odl_license_tag, 'dcterms:format')
+            full_content_type = subtag(odl_license_tag, 'dcterms:format')
+
+            # By default, dcterms:format includes the media type of a
+            # DRM-free resource.
+            content_type = full_content_type
             drm_schemes = []
-            protection_tags = parser._xpath(odl_license_tag, 'odl:protection') or []
+
+            # But it may instead describe an audiobook protected with
+            # the Feedbooks access-control scheme.
+            feedbooks_audio = "%s; protection=%s"  % (
+                MediaTypes.AUDIOBOOK_MANIFEST_MEDIA_TYPE,
+                DeliveryMechanism.FEEDBOOKS_AUDIOBOOK_DRM
+            )
+            if full_content_type == feedbooks_audio:
+                content_type = MediaTypes.AUDIOBOOK_MANIFEST_MEDIA_TYPE
+                drm_schemes.append(DeliveryMechanism.FEEDBOOKS_AUDIOBOOK_DRM)
+
+            # Additional DRM schemes may be described in <odl:protection>
+            # tags.
+            protection_tags = parser._xpath(
+                odl_license_tag, 'odl:protection'
+            ) or []
             for protection_tag in protection_tags:
                 drm_scheme = subtag(protection_tag, 'dcterms:format')
+                if drm_scheme:
+                    drm_schemes.append(drm_scheme)
+
+            for drm_scheme in (drm_schemes or [None]):
+                formats.append(
+                    FormatData(
+                        content_type=content_type,
+                        drm_scheme=drm_scheme,
+                        rights_uri=RightsStatus.IN_COPYRIGHT,
+                    )
+                )
```

Comment:
This is the same code as below, just refactored to remove the `if not drm_schemes` path.